### DISCUSSION TITLE
Asynchronous support

### DISCUSSION THREAD

**Discussion by magiskboy** [fastapi/sqlmodel#597] [Questions]
Does it support asynchronous api?
So if it doesn't support async api, is view function in fastapi blocked if I use sqlmodel api?

**Comment by antont** [comment]
async is supported. there are several issues here that discuss it and have examples, was it added to the docs recently too?

**Reply by matanper** [reply]
Hey @antont, 
I couldn't find the examples you are talking about. I see that `sqlmodel` wraps `create_engine` but not `create_async_engine`.
Is it mandatory to use the wrapped `sqlmodel.create_engine` or I can use directly sqlalchemy?

**Reply by elefher** [reply]
Hey @matanper , a bit late reply, but from some experiments I ran you can use  create_async_engine from sqlalchemy directly along with sqlmodel. Now, if you do not want to use it like this I see that sqlmodel.create_engine supports _is_aynch=True param, but I haven't really tested this approach yet
